build fail real (you just need to update the test).
```
======================================================================
FAIL: test_instantiate_all (testapps.test_pillowtop.tests.test_settings.PillowtopSettingsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/testapps/test_pillowtop/tests/test_settings.py", line 35, in test_instantiate_all
    self.assertEqual(len(all_pillow_configs), len(expected_meta))
AssertionError: 49 != 48
----------------------------------------------------------------------
```

--- a/corehq/ex-submodules/pillowtop/processors/form.py
+++ b/corehq/ex-submodules/pillowtop/processors/form.py
@@ -24,16 +24,16 @@ class AppFormSubmissionTrackerProcessor(PillowProcessor):
         domain = change.metadata.domain
 
         if build_id and domain:
-            app = self._get_app(domain, build_id)
-
-            if not app.has_submissions:
-                app.has_submissions = True
-                app.save()
-                self._get_app.clear(self, domain, build_id)
+            self._mark_has_submission(domain, build_id)
 
     @quickcache(['domain', 'build_id'], timeout=60 * 60)
-    def _get_app(self, domain, build_id):
+    def _mark_has_submission(self, domain, build_id):
+        app = None
         try:
-            return get_app(domain, build_id)
+            app = get_app(domain, build_id)
         except Http404:
-            return None
+            pass
+
+        if app and not app.has_submissions:
+            app.has_submissions = True
+            app.save()

--- a/testapps/test_pillowtop/tests/data/all-pillow-meta.json
+++ b/testapps/test_pillowtop/tests/data/all-pillow-meta.json
@@ -602,7 +602,7 @@
     "AppFormSubmissionTrackerPillow": {
         "advertised_name": "AppFormSubmissionTrackerPillow",
         "change_feed_type": "KafkaChangeFeed",
-        "checkpoint_id": "AppFormSubmissionTrackerPillow",
+        "checkpoint_id": "app-form-submission-tracker",
         "full_class_name": "pillowtop.pillow.interface.ConstructedPillow",
         "name": "AppFormSubmissionTrackerPillow"
     },

--- a/testapps/test_pillowtop/tests/data/all-pillow-meta.json
+++ b/testapps/test_pillowtop/tests/data/all-pillow-meta.json
@@ -599,6 +599,13 @@
         "name": "XFormPillow",
         "unique_id": "test_xforms_2016-06-09"
     },
+    "AppFormSubmissionTrackerPillow": {
+        "advertised_name": "AppFormSubmissionTrackerPillow",
+        "change_feed_type": "KafkaChangeFeed",
+        "checkpoint_id": "AppFormSubmissionTrackerPillow",
+        "full_class_name": "pillowtop.pillow.interface.ConstructedPillow",
+        "name": "AppFormSubmissionTrackerPillow"
+    },
     "kafka-ucr-main": {
         "advertised_name": "kafka-ucr-main",
         "change_feed_type": "KafkaChangeFeed",

